File: lesson_secure_coding/py/broken_access_control_fix.py
import http.server
import socketserver


TOKEN = 'valid_admin_token'


class MyHandler(http.server.SimpleHTTPRequestHandler):
    def do_GET(self):
        if self.path == "/admin":
            if self.headers.get("Authorization") == f"Bearer {TOKEN}":
                self.send_response(200)
                self.end_headers()
                self.wfile.write(b"Welcome to the admin page!")
            else:
                self.send_response(403)
                self.end_headers()
                self.wfile.write(b"Access denied.")
        else:
            self.send_response(404)
            self.end_headers()
            self.wfile.write(b"Page not found.")


PORT = 8000
with socketserver.TCPServer(("", PORT), MyHandler) as httpd:
    print(f"Serving on port {PORT}")
    httpd.serve_forever()

File: lesson_secure_coding/node/broken_access_control_fix.js
const http = require('http')
const validAdminToken = "valid_admin_token";

const server = http.createServer((req, res) => {
    if (req.url === '/admin') {
        const authHeader = req.headers['authorization'];
        if (authHeader === `Bearer ${validAdminToken}`) {
            res.writeHead(200, { 'Content-Type': 'text/plain' });
            res.end('Welcome to the admin page!');
        } else {
            res.writeHead(403, { 'Content-Type': 'text/plain' });
            res.end('Access denied.');
        }
    } else {
        res.writeHead(404, { 'Content-Type': 'text/plain' });
        res.end('Page not found.');
    }
});

server.listen(8000, () => {
    console.log('Serving on port 8000');
});
